src/platforms/hosted/remote_jtagtap.c

```diff
@@ -126,26 +126,19 @@ static void jtagtap_tdi_tdo_seq(
 			chunk = 64;
 		}
 		ticks -= chunk;
-		uint8_t di[8];
-		memset(di, 0, 8);
+		uint64_t di = 0;
 		int bytes = (chunk + 7) >> 3;
-		if (DI) {
-			memcpy(&di, DI, bytes);
-			int remainder = chunk & 7;
-			DI += bytes;
-			DI += bytes;
-			if (remainder) {
-				uint8_t rem = *DI;
-				rem &= (1 << remainder) - 1;
-				*di = rem;
-			}
+		int i = 0;
+		for (; i < bytes; i++) {
+			di = di | (*DI << (i * 8));
```

Comment:
`|=` is a thing - as you aren't mucking with 'di' any other way, please use that instead or clarity and safety

Reply:
Please explain why |=  has more clarity or  safety.

Reply:
you are or'ing a value to `di` and making it equal to that new value, that is literally what `|=` means and has the benefit that you can't typo the left hand side of the `|` or otherwise accidentally modify the value in some other way first.

Reply:
Changed as requested.